File: python/examples.py
import maze.generator.RecursiveBacktracker
import maze.generator.BinaryTree
import maze.generator.Sidewinder
import maze.generator.AldousBroder
import maze.generator.HuntKill
import maze.generator.Wilsons
import maze.generator.Prims
from maze.solver.Dijkstras import path_to_goal

# height and width of the generated mazes
height = 10
width = 15

# Maze Generation Algorithms

# generate a random maze using BinaryTree algorithm
bt_maze = maze.generator.BinaryTree.generate(height, width)
print(f"binary tree {height}x{width}")
print(f"{bt_maze}\n\n")

# generate a random maze using Sidewinder algorithm
sw_maze = maze.generator.Sidewinder.generate(height, width)
print(f"sidewinder {height}x{width}")
print(f"{sw_maze}\n\n")

# generate a random maze using Aldous-Broder algorithm
ab_maze = maze.generator.AldousBroder.generate(height, width)
print(f"Aldous-Broder {height}x{width}")
print(f"{ab_maze}\n\n")

# generate a random maze using Hunt and Kill algorithm
hk_maze = maze.generator.HuntKill.generate(height, width)
print(f"Hunt and Kill {height}x{width}")
print(f"{hk_maze}\n\n")

# generate a random maze using Prims algorithm
pr_maze = maze.generator.Prims.generate(height, width)
print(f"prims {height}x{width}")
print(f"{pr_maze}\n\n")

# generate a random maze using Wilson's algorithm
w_maze = maze.generator.Wilsons.generate(height, width)
print(f"Wilsons {height}x{width}")
print(f"{w_maze}\n\n")

# generate a braided maze, from the Wilson's maze, by removing 50% of the dead-ends
w_maze.braid(0.5)
print(f"Braided maze (from Wilson's) with 50% dead-ends removed {height}x{width}")
print(f"{w_maze}\n\n")

# generate a random maze using RecursiveBacktracking algorithm
rb_maze = maze.generator.RecursiveBacktracker.generate(height, width)
print(f"recursive backtracker {height}x{width}")
print(f"{rb_maze}\n\n")

# Finding the shortest path from a starting cell to a goal cell

start = rb_maze.get(0, 0)                               # north-west corner
goal = rb_maze.get(rb_maze.rows - 1, rb_maze.cols - 1)  # south-east corner
shortest_path = path_to_goal(start, goal)
print(f"shortest path in recursive backtracker {height}x{width}")
print(rb_maze.print_grid(shortest_path))

File: java/src/main/java/com/mazes/Examples.java
package com.mazes;

import com.mazes.generator.*;
import com.mazes.solver.Dijkstras;

public class Examples {


    public static void main(String[] args) {
        int height = 10;
        int width = 15;

        // try to parse the height and width from STDIN, if provided
        if (args.length > 1) {
            height = Integer.parseInt(args[1]);
        }

        if (args.length > 2) {
            width = Integer.parseInt(args[2]);
        }

        /// Maze Generation Algorithms
        Grid abMaze = new AldousBroder().generate(height, width);
        System.out.println("Aldous Broder " + height + "x" + width);
        System.out.println(abMaze + "\n\n");

        Grid btMaze = new BinaryTree().generate(height, width);
        System.out.println("Binary Tree " + height + "x" + width);
        System.out.println(btMaze + "\n\n");

        Grid hkMaze = new HuntKill().generate(height, width);
        System.out.println("Hunt and Kill " + height + "x" + width);
        System.out.println(hkMaze + "\n\n");

        Grid pMaze = new Prims().generate(height, width);
        System.out.println("Prims " + height + "x" + width);
        System.out.println(pMaze + "\n\n");

        Grid rbMaze = new RecursiveBacktracker().generate(height, width);
        System.out.println("Recursive Backtracker " + height + "x" + width);
        System.out.println(rbMaze + "\n\n");

        Grid swMaze = new Sidewinder().generate(height, width);
        System.out.println("Sidewinder " + height + "x" + width);
        System.out.println(swMaze + "\n\n");

        Grid wMaze = new Wilsons().generate(height, width);
        System.out.println("Wilsons " + height + "x" + width);
        System.out.println(wMaze + "\n\n");

        // remove 50% of dead-ends, which will create loops in the maze, a.k.a a "braided" maze
        wMaze.braid(0.5f);
        System.out.println("Wilsons Braided (50% dead-ends removed) " + height + "x" + width);
        System.out.println(wMaze + "\n\n");

        // find the shortest path from north-west corner to south-east corner
        Cell nwCell = wMaze.get(0, 0);
        Cell seCell = wMaze.get(wMaze.rows - 1, wMaze.cols - 1);
        Distances shortestPath = Dijkstras.pathToGoal(nwCell, seCell);
        System.out.println("Shortest path from north-west corner to south-east corner");
        System.out.println(wMaze.printDistances(shortestPath) + "\n\n");
    }
}
